reference/directory-settings.md
# Create a Directory Settings List

Compose parts of and complete directory settings controls lists.  
  
Most users will not want or need to change the directory folders and
file names, but it is helpful to have them be flexible for certain
circumstances, and `directory_files`, `directory_subdirectories`, and
`directory_settings` gather them into a set of lists for pipeline
functionality.  
`directory_resources` dictates the `source` and `version` (and any other
specialized arguments) for each of the raw resources collected for the
directory.  
`time_settings` collates the temporal aspects of the data, and can be
used to set up alternate forecast timelines for specialized projects and
sandboxing.

## Usage

``` r
directory_settings(
  files = directory_files(),
  subdirectories = directory_subdirectories(),
  resources = directory_resources(),
  time = time_settings(),
  confidence_level = 0.95,
  nsamples = 10000,
  save = TRUE,
  overwrite = TRUE,
  force = FALSE,
  unzip_pause = 30,
  download_timeout = getOption("timeout")
)

time_settings(
  timeseries_start = as.Date("1995-01-01"),
  origin = Sys.Date(),
  forecast_date = Sys.Date(),
  lead_time = 365,
  max_lag = 365,
  lag_buffer = 60,
  lead_time_buffer = 30
)

directory_files(
  directory_configuration = "directory_configuration.yaml",
  app = "app.R",
  newmoons = "newmoons.csv",
  covariates = "covariates.csv",
  datasets_controls = "datasets_controls.yaml",
  models_controls = "models_controls.yaml",
  forecasts_evaluations = "forecasts_evaluations.csv",
  forecasts_results = "forecasts_results.csv",
  forecasts_metadata = "forecasts_metadata.csv",
  metadata = "metadata.yaml",
  about_md = "about.md",
  models_html = "models.html",
  models_rmd = "models.Rmd",
  rodents_profiles_html = "rodents_profiles.html",
  rodents_profiles_csv = "rodents_profiles.csv",
  species_names = "species_names.csv"
)

directory_subdirectories(
  forecasts = "forecasts",
  fits = "fits",
  models = "models",
  resources = "resources",
  data = "data",
  www = "www"
)

directory_resources(
  PortalData = list(source = "github", version = "latest"),
  portalPredictions = list(source = "github", version = NULL),
  climate_forecasts = list(source = "NMME", version = as.character(Sys.Date()), data =
    list(mintemp = "tasmin", meantemp = "tasmean", maxtemp = "tasmax", precipitation =
    "pr"))
)

production_settings(download_timeout = max(getOption("timeout"), 600))

sandbox_settings()
```

## Arguments

- files:

  `list` of `character` names of standard files, see `directory_files`.

- subdirectories:

  `list` of `character` names of standard subdirectories, see
  `directory_subdirectories`.

- resources:

  `character` name for the resources subdirectory.

- time:

  `list` of time settings, see `time_settings`.

- confidence_level:

  `numeric` confidence level used in summarizing model output. Must be
  between `0` and `1`.

- nsamples:

  `integer` (or integer `numeric`) number of samples used to summarizing
  model output of sample-based estimates.

- save:

  `logical` indicator controlling if the output should be saved out.

- overwrite:

  `logical` indicator of whether or not file writing should occur even
  if a local copy already exists.

- force:

  `logical` indicator of whether or not existing files or folders (such
  as the archive) should be over-written if an up-to-date copy exists
  (most users should leave as `FALSE`).

- unzip_pause:

  Positive `integer` or integer `numeric` seconds for pausing during
  steps around unzipping that require time delay.

- download_timeout:

  Positive `integer` or integer `numeric` seconds for timeout on
  downloads. Temporarily overrides the `"timeout"` option in
  [`base::options`](https://rdrr.io/r/base/options.html).

- timeseries_start:

  `Date` after which historic samples are included in the timeseries
  fit. Default value is `1995-01-01`, corresponding to moon 217.

- origin:

  `Date` forecast origin. Default is today's date (set using
  [`Sys.Date`](https://rdrr.io/r/base/Sys.time.html)).

- forecast_date:

  `Date` of when the forecasts are occurring. Default is today's date
  (set using [`Sys.Date`](https://rdrr.io/r/base/Sys.time.html)).

- lead_time:

  `integer` (or integer `numeric`) value for the number of calendar days
  forward a forecast will cover.  
  As of version 0.51.0, default is now `365`, which when divided by 29.5
  (duration of a lunar month), gives 13. The previous value was
  previously 12. We are now using 13 to align with the timestep being a
  lunar month, and 13 lunar months covers a full calendar year.

- max_lag:

  `integer` (or integer `numeric`) maximum number of calendar days that
  any covariate is lagged for prediction in a model.  
  Default is `365` for the logistic covariate models.

- lag_buffer:

  `integer` (or integer `numeric`) additional number of calendar days
  back in time to add to the maximum lag.  
  Default value of `60` corresponds to two additional lunar months.

- lead_time_buffer:

  `integer` (or integer `numeric`) additional number of calendar days
  forward in time to forecast.  
  Default value of `30` corresponds to one additional lunar month.

- directory_configuration:

  `character` name for the directory configuration YAML.

- app:

  `character` name for the application R file.

- newmoons:

  `character` name for the lunar data csv.

- covariates:

  `character` name for the combined historical and forecast covariates
  csv.

- datasets_controls:

  `character` name for the YAML of datasets control list(s).

- models_controls:

  `character` name for the YAML of models controls list(s).

- forecasts_evaluations:

  `character` name for the forecast evaluations csv.

- forecasts_results:

  `character` name for the forecast combination results csv.

- forecasts_metadata:

  `character` name for the forecast metadata csv.

- metadata:

  `character` name for the Forecast metadata YAML.

- about_md:

  `character` name for the md file for the about page of the app.

- models_html:

  `character` name for the html file where the models page will be saved
  for the app.

- models_rmd:

  `character` name for the Rmd file for the models page of the app.

- rodents_profiles_html:

  `character` name for the html file where the rodents profiles page
  will be saved for the app.

- rodents_profiles_csv:

  `character` name for the csv file containing content for the rodents
  profiles table for the app.

- species_names:

  `character` name for the csv file containing the output from
  [`portalr::rodent_species`](https://weecology.github.io/portalr/reference/rodent_species.html),
  with `set = "forecasting"`, `type = "table"`, and `total = TRUE`.

- forecasts:

  `character` name for the forecasts subdirectory.

- fits:

  `character` name for the fits subdirectory.

- models:

  `character` name for the models subdirectory.

- data:

  `character` name for the data subdirectory.

- www:

  `character` name for the application www helpers subdirectory.

- PortalData:

  `list` of `source` and `version` elements of `character` values for
  the Portal Data download. Default values retrieve the latest data from
  github

- portalPredictions:

  `list` of `source` and `version` elements of `character` values for
  the archive download. Default values point to github, but
  `verison = NULL` indicates no download.

- climate_forecasts:

  `list` of `source`, `version`, and `data` elements of `character`
  values for the climate forecasts download. Default values retrieve the
  current day's forecast of min, mean, and max temperature and
  precipitation from the Northwest Knowledge Network's North American
  Multi-Model Ensemble (NMME) climate forecasts.

## Value

Named `list` of settings for the directory (for `directory_settings`) or
`list` of settings components (for `directory_files`,
`directory_subdirectories`, and `directory_resources`).

## See also

Directory orchestration functions:
[`directory configuration file`](https://weecology.github.io/portalcasting/reference/directory-configuration-file.md),
[`directory creation`](https://weecology.github.io/portalcasting/reference/directory-creation.md),
[`directory filling`](https://weecology.github.io/portalcasting/reference/directory-filling.md),
[`directory paths`](https://weecology.github.io/portalcasting/reference/directory-paths.md)

## Examples

``` r
   directory_settings( )
   directory_files( )
   directory_subdirectories( )
   directory_resources( )
   time_settings( )
```


reference/directory-configuration-file.md
# Create, Update, and Read the Directory Configuration File

The directory configuration file is a special file within the directory
setup and has its own set of functions.  
  
`write_directory_configuration` creates the YAML metadata configuration
file. It is (and should only be) called from within
[`setup_dir`](https://weecology.github.io/portalcasting/reference/directory-creation.md),
as it captures information about the compute environment used to
instantiate the directory.  
  
`read_directory_configuration` reads the YAML config file into the R
session.  
  
`read_directory_settings` reads the YAML config file into the R session
and pulls just the directory settings list in.

## Usage

``` r
write_directory_configuration(
  main = ".",
  settings = directory_settings(),
  quiet = FALSE,
  verbose = FALSE
)

read_directory_configuration(main = ".")

read_directory_settings(main = ".")

update_directory_configuration(main = ".")
```

## Arguments

- main:

  `character` value of the name of the main component of the directory
  tree.

- settings:

  `list` of controls for the directory, with defaults set in
  [`directory_settings`](https://weecology.github.io/portalcasting/reference/directory-settings.md).

- quiet:

  `logical` indicator if progress messages should be quieted.

- verbose:

  `logical` indicator of whether or not to print out all of the
  messages.

## Value

`list` of directory configurations,
[`invisible`](https://rdrr.io/r/base/invisible.html)-ly.

## See also

Directory orchestration functions:
[`directory creation`](https://weecology.github.io/portalcasting/reference/directory-creation.md),
[`directory filling`](https://weecology.github.io/portalcasting/reference/directory-filling.md),
[`directory paths`](https://weecology.github.io/portalcasting/reference/directory-paths.md),
[`directory settings`](https://weecology.github.io/portalcasting/reference/directory-settings.md)

File read-write functions:
[`read write data`](https://weecology.github.io/portalcasting/reference/read-write-data.md)

## Examples

``` r
if (FALSE) { # \dontrun{
   main1 <- file.path(tempdir(), "standard")
   setup_dir(main = main1)

   settings1 <- read_directory_settings(main = main1)
   config1   <- read_directory_configuration(main = main1)

   unlink(main1, recursive = TRUE)
} # }
```


NEWS.md
# portalcasting (development version)

Version numbers follow [Semantic Versioning](https://semver.org/).


# [portalcasting 0.60.6](https://github.com/weecology/portalcasting/releases/tag/v0.60.6)
*08-21-2024*
## Bug fix on evaluation.
* Add sNaiveArima, sAutoArima models. The models were skipped.

# [portalcasting 0.60.5](https://github.com/weecology/portalcasting/releases/tag/v0.60.5)
*01-01-2024*
## Fix Zipping function.
* Avoid creating extra folders from relative paths

# [portalcasting 0.60.4](https://github.com/weecology/portalcasting/releases/tag/v0.60.4)
*01-01-2024*

## Zip resources forecast
* Zipping forecast folder in resources

# [portalcasting 0.60.3](https://github.com/weecology/portalcasting/releases/tag/v0.60.3)
*12-16-2023*

## Add tests
* Tests for zip and unszip functions

# [portalcasting 0.60.2](https://github.com/weecology/portalcasting/releases/tag/v0.60.2)
*11-16-2023*

## Repository moved
* Store forecasts by date using the zip format
* Move from portalPredictions to portal-forecasts due to size limitations.

# [portalcasting 0.60.1](https://github.com/weecology/portalcasting/releases/tag/v0.60.1)
*5-22-2023*

## Minor doc edits

# [portalcasting 0.60.0](https://github.com/weecology/portalcasting/releases/tag/v0.60.0)
*5-19-2023*

## Automating editing of dockerhub description
* Using a [3rd party action](https://github.com/peter-evans/dockerhub-description) while docker gets this functionality integrated 
* addresses #361

# [portalcasting 0.59.0](https://github.com/weecology/portalcasting/releases/tag/v0.59.0)
*5-18-2023*

## Including curl in the Dockerfile
* thought it was installed, but it isn't in tidyverse, just verse

# [portalcasting 0.58.0](https://github.com/weecology/portalcasting/releases/tag/v0.58.0)
*5-18-2023*

## Editing documentation
* Auto templates for issues and PRs 

## Including git2r in the package
* Was removed accidentally from dockerfile causing breaks in main repo, seeing if it works to be in the package itself

# [portalcasting 0.57.0](https://github.com/weecology/portalcasting/releases/tag/v0.57.0)
*5-12-2023*

## Editing documentation
* Tightening and improving language in Title and Description fields in DESCRIPTION file

## Preventing duplicate keys in control lists at the top level

## Adding docker images with setup done
* including a production and a sandbox image

## Testing docker images prior to pushing them
* A simple "can you run `library(portalcasting)`" to start, can use this entry point to expand the testing suite on the image in the future


# portalcasting 0.56.0
*5-5-2023*

## Tidying the app.R script
* Facilitating the working across different calling styles.

# [portalcasting 0.55.0](https://github.com/weecology/portalcasting/releases/tag/v0.55.0)
*5-4-2023*

## Reorganizing the app pointing
* Now having a single app.R file in the top level of the directory simplifies the issue around where the files are for the app
* This replaces the global.R, ui.R, and server.R files 

# [portalcasting 0.54.0](https://github.com/weecology/portalcasting/releases/tag/v0.54.0)
*5-3-2023*

## Adding `sandbox` and `production` version of `update_` functions. 

## Adding Pat as an author

## Pointing the dockerfile to the current build SHA
* Previously, we were using default settings for `remotes::install_github`, which actually points to HEAD, when testing the build and push for the docker action. that was fine (although not great) when we were only running that action on tagged version releases, as the main branch was typically up-to-date with the tag and such, but this isn't exact and it also means any testing of the build on a PR was actually still grabbing from main, which is not what we want.
* We now use the SHA for the specific event that triggers the build by passing it into the docker file as an ARG

# [portalcasting 0.53.0](https://github.com/weecology/portalcasting/releases/tag/v0.53.0)
*4-26-2023*

## Removing temporary patch for backwards compatibility  

# [portalcasting 0.52.0](https://github.com/weecology/portalcasting/releases/tag/v0.52.0)
*4-25-2023*

## Patching evaluations issues
* The table was too big as was for github standard, so removing columns of repeated content that can be grabbed from metadata

# [portalcasting 0.51.1](https://github.com/weecology/portalcasting/releases/tag/v0.51.1)
*4-24-2023*

## Patches to pkgdown site


# [portalcasting 0.51.0](https://github.com/weecology/portalcasting/releases/tag/v0.51.0)
*4-24-2023*

## Major updates to pkgdown site
* Shift to bootstrap 5

## Include CI tests of \dontrun examples and eval=FALSE vignettes
* Because of the long run time of some of the code, we wrap most documentation examples and vignette code in a way to prevent evaluation in real-time. As a result, much of the documentation code isn't run and therefore would not break builds if it would error. 
* To address this, we add two scripts in the new inst/extra_testing folder and a github action runner for each.

## Added CITATION file
* Cites the JOSS paper.

## Bringing the app code into the package
* Improves robustness of building the app (includes code and dependencies in the docker image, allows for unit testing app components, etc.)
* Also allows users to spin-up a local version with `run_web_app()` pointed to `main`

## Using `arrow` to speed up reading and writing files

## Evaluation figures now read from evaluations file
* Avoids computing evaluations while generating plots

## Elimination of model-named functions
* The models are now implemented using `cast` on their fit and cast elements in their control lists
* Only models the need new functions have them (`meta_tsglm` and `fit_runjags`) for fitting
* `forecast` method used generally now for `cast` function
  * introduction of `forecast` methods for `tsglm` and `runjags` objects

## Shifting covariates to a daily-level build initial step

R/settings.R
#' @title Create a Directory Settings List
#'
#' @description Compose parts of and complete directory settings controls lists. \cr \cr
#'              Most users will not want or need to change the directory folders and file names, but it is helpful to have them be flexible for certain circumstances, and `directory_files`, `directory_subdirectories`, and `directory_settings` gather them into a set of lists for pipeline functionality. \cr
#'              `directory_resources` dictates the `source` and `version` (and any other specialized arguments) for each of the raw resources collected for the directory. \cr
#'              `time_settings` collates the temporal aspects of the data, and can be used to set up alternate forecast timelines for specialized projects and sandboxing. 
#'
#' @param files `list` of `character` names of standard files, see [`directory_files`].
#'
#' @param subdirectories `list` of `character` names of standard subdirectories, see [`directory_subdirectories`].
#'
#' @param resources `list` of `list`s for standard resources, see [`directory_resources`].
#'
#' @param time `list` of time settings, see [`time_settings`].
#'
#' @param newmoons `character` name for the lunar data csv.
#'
#' @param covariates `character` name for the combined historical and forecast covariates csv.
#'
#' @param forecasts_metadata `character` name for the forecast metadata csv.
#'
#' @param metadata `character` name for the Forecast metadata YAML.
#'
#' @param directory_configuration `character` name for the directory configuration YAML.
#'
#' @param datasets_controls `character` name for the YAML of datasets control list(s).
#'
#' @param models_rmd `character` name for the Rmd file for the models page of the app.
#'
#' @param models_html `character` name for the html file where the models page will be saved for the app.
#'
#' @param about_md `character` name for the md file for the about page of the app.
#'
#' @param rodents_profiles_html `character` name for the html file where the rodents profiles page will be saved for the app.
#'
#' @param rodents_profiles_csv `character` name for the csv file containing content for the rodents profiles table for the app.
#'
#' @param species_names `character` name for the csv file containing the output from [`portalr::rodent_species`], with `set = "forecasting"`, `type = "table"`, and `total = TRUE`.
#'
#' @param models_controls `character` name for the YAML of models controls list(s).
#'
#' @param forecasts_evaluations `character` name for the forecast evaluations csv.
#'
#' @param forecasts_results `character` name for the forecast combination results csv.
#'
#' @param resources `character` name for the resources subdirectory.
#'
#' @param data `character` name for the data subdirectory.
#'
#' @param models `character` name for the models subdirectory.
#'
#' @param fits `character` name for the fits subdirectory.
#'
#' @param forecasts `character` name for the forecasts subdirectory.
#'
#' @param www `character` name for the application www helpers subdirectory.
#'
#' @param app `character` name for the application R file.
#'
#' @param PortalData `list` of `source` and `version` elements of `character` values for the Portal Data download. Default values retrieve the latest data from github
#'
#' @param portalPredictions `list` of `source` and `version` elements of `character` values for the archive download. Default values point to github, but `verison = NULL` indicates no download.
#'
#' @param climate_forecasts `list` of `source`, `version`, and `data` elements of `character` values for the climate forecasts download. Default values retrieve the current day's forecast of min, mean, and max temperature and precipitation from the Northwest Knowledge Network's North American Multi-Model Ensemble (NMME) climate forecasts.
#'
#' @param save `logical` indicator controlling if the output should be saved out.
#'
#' @param overwrite `logical` indicator of whether or not file writing should occur even if a local copy already exists.
#'
#' @param force `logical` indicator of whether or not existing files or folders (such as the archive) should be over-written if an up-to-date copy exists (most users should leave as `FALSE`).
#'
#' @param unzip_pause Positive `integer` or integer `numeric` seconds for pausing during steps around unzipping that require time delay.
#'
#' @param download_timeout Positive `integer` or integer `numeric` seconds for timeout on downloads. Temporarily overrides the `"timeout"` option in [`base::options`].
#'
#' @param origin `Date` forecast origin. Default is today's date (set using [`Sys.Date`]).
#'
#' @param forecast_date `Date` of when the forecasts are occurring. Default is today's date (set using [`Sys.Date`]).
#'
#' @param timeseries_start `Date` after which historic samples are included in the timeseries fit. Default value is `1995-01-01`, corresponding to moon 217.
#'
#' @param lead_time `integer` (or integer `numeric`) value for the number of calendar days forward a forecast will cover. \cr
#'   As of version 0.51.0, default is now `365`, which when divided by 29.5 (duration of a lunar month), gives 13. The previous value was previously 12. We are now using 13 to align with the timestep being a lunar month, and 13 lunar months covers a full calendar year.
#'
#' @param max_lag `integer` (or integer `numeric`) maximum number of calendar days that any covariate is lagged for prediction in a model. \cr
#'   Default is `365` for the logistic covariate models.
#'
#' @param lag_buffer `integer` (or integer `numeric`) additional number of calendar days back in time to add to the maximum lag. \cr
#'   Default value of `60` corresponds to two additional lunar months. 
#'
#' @param lead_time_buffer `integer` (or integer `numeric`) additional number of calendar days forward in time to forecast. \cr
#'   Default value of `30` corresponds to one additional lunar month.
#'
#' @param confidence_level `numeric` confidence level used in summarizing model output. Must be between `0` and `1`.
#'
#' @param nsamples `integer` (or integer `numeric`) number of samples used to summarizing model output of sample-based estimates.
#'
#' @return Named `list` of settings for the directory (for `directory_settings`) or `list` of settings components (for `directory_files`, `directory_subdirectories`, and `directory_resources`).
#'
#' @name directory settings
#'
#' @aliases settings
#'
#' @family orchestration
#' @family customization
#'
#' @examples
#'    directory_settings( )
#'    directory_files( )
#'    directory_subdirectories( )
#'    directory_resources( )
#'    time_settings( )
#'
#'
NULL

#' @rdname directory-settings
#'
#' @export
#'
directory_settings <- function (files             = directory_files( ),
                                subdirectories    = directory_subdirectories( ),
                                resources         = directory_resources( ),
                                time              = time_settings( ),
                                confidence_level  = 0.95,
                                nsamples          = 1e4,
                                save              = TRUE,
                                overwrite         = TRUE,
                                force             = FALSE,
                                unzip_pause       = 30,
                                download_timeout  = getOption("timeout")) {

  list(files            = files,
       subdirectories   = subdirectories,
       resources        = resources,
       repository       = "portal-forecasts",
       confidence_level = confidence_level,
       nsamples         = nsamples,
       time             = time,
       save             = save,
       force            = force,
       overwrite        = overwrite,
       unzip_pause      = unzip_pause,
       download_timeout = download_timeout)

}

#' @rdname directory-settings
#'
#' @export
#'
time_settings <- function (timeseries_start = as.Date("1995-01-01"),
                           origin           = Sys.Date( ),
                           forecast_date    = Sys.Date( ),
                           lead_time        = 365,
                           max_lag          = 365,
                           lag_buffer       = 60,
                           lead_time_buffer = 30) {


  timeseries_start_lagged <- timeseries_start - max_lag - lag_buffer
  forecast_start          <- origin + 1
  forecast_end            <- origin + lead_time
  forecast_end_buffered   <- origin + lead_time + lead_time_buffer

  list(timeseries_start        = as.character(timeseries_start),
       timeseries_start_lagged = as.character(timeseries_start_lagged),
       forecast_start          = as.character(forecast_start),
       forecast_end            = as.character(forecast_end),
       forecast_end_buffered   = as.character(forecast_end_buffered),
       origin                  = as.character(origin),
       forecast_date           = as.character(forecast_date),
       lead_time               = lead_time,
       lead_time_buffer        = lead_time_buffer,
       max_lag                 = max_lag,
       lag_buffer              = lag_buffer)

}

#' @rdname directory-settings
#'
#' @export
#'
directory_files <- function (directory_configuration = "directory_configuration.yaml",
                             app                     = "app.R",
                             newmoons                = "newmoons.csv",
                             covariates              = "covariates.csv",
                             datasets_controls       = "datasets_controls.yaml",
                             models_controls         = "models_controls.yaml",
                             forecasts_evaluations   = "forecasts_evaluations.csv",
                             forecasts_results       = "forecasts_results.csv",
                             forecasts_metadata      = "forecasts_metadata.csv",
                             metadata                = "metadata.yaml",
                             about_md                = "about.md",
                             models_html             = "models.html",
                             models_rmd              = "models.Rmd",
                             rodents_profiles_html   = "rodents_profiles.html",
                             rodents_profiles_csv    = "rodents_profiles.csv",
                             species_names           = "species_names.csv") {

  list(directory_configuration   = directory_configuration,
       app                       = app,
       newmoons                  = newmoons,
       covariates                = covariates,
       datasets_controls         = datasets_controls,
       models_controls           = models_controls,
       forecasts_evaluations     = forecasts_evaluations,
       forecasts_results         = forecasts_results,
       forecasts_metadata        = forecasts_metadata,
       about_md                  = about_md,
       models_rmd                = models_rmd,
       models_html               = models_html,
       rodents_profiles_html     = rodents_profiles_html,
       rodents_profiles_csv      = rodents_profiles_csv,
       species_names             = species_names,
       metadata                  = metadata)

}


#' @rdname directory-settings
#'
#' @export
#'
directory_subdirectories <- function (forecasts = "forecasts",
                                      fits      = "fits",
                                      models    = "models",
                                      resources = "resources",
                                      data      = "data",
                                      www       = "www") {

  list(forecasts = forecasts,
       fits      = fits,
       models    = models,
       resources = resources,
       data      = data,
       www       = www)

}

#' @rdname directory-settings
#'
#' @export
#'
directory_resources <- function (PortalData         = list(source  = "github",
                                                           version = "latest"),
                                 portalPredictions  = list(source  = "github",
                                                           version = NULL),
                                 climate_forecasts  = list(source  = "NMME",
                                                           version = as.character(Sys.Date()),
                                                           data    = list(mintemp       = "tasmin",
                                                                          meantemp      = "tasmean",
                                                                          maxtemp       = "tasmax",
                                                                          precipitation = "pr"))) {

  list(PortalData         = PortalData,
       portalPredictions  = portalPredictions,
       climate_forecasts  = climate_forecasts)

}


#' @rdname directory-settings
#'
#' @export
#'
production_settings <- function (download_timeout  = max(getOption("timeout"), 600)) {

  resources <- directory_resources(portalPredictions = list(source  = "github",
                                                            version = "latest"))

  directory_settings(resources        = resources,
                     download_timeout = download_timeout)

}


#' @rdname directory-settings
#'
#' @export
#'
sandbox_settings <- function () {

  directory_settings( )

}






